### DISCUSSION TITLE
Nested inline styles

### DISCUSSION THREAD

**Discussion by nborzunov** [chakra-ui/chakra-ui#5994] [Q&A]
Is there any ways to provide nested inline styles to ChakraUI component?

With CSS styles it works well
```
<NavigationButton
    icon={Icons.Settings}
    label={t('settings')}
    as={MenuButton}
    className='menu-button'
/>
```

styles.css
```
.menu-button span {
	display: flex;
}
```
But I want to make it more clear and style inside jsx

Is it possible to do something like that?

```
<NavigationButton
    icon={Icons.Settings}
    label={t('settings')}
    as={MenuButton}
    css={{
        span: {
            display: 'flex,
        }
    }}
/>
```

But actually doing this way I overwrite all NavigationButton styles, such as variant, colorScheme, etc.
So I just need to extend component style, without extending theme and also without additional CSS files.

**Comment by BillyPurvis** [comment]
Hello! 

You should be able to do this using the [sx props](https://chakra-ui.com/docs/styled-system/features/the-sx-prop#creating-nested-selectors). Similarly, you can use [layerStyles](https://chakra-ui.com/docs/styled-system/features/text-and-layer-styles#layer-style) to achieve this if you'd like to re-use this set of styles, but not augment / add variants to a specific component.  


Example block:

```
<Box
  mt={2}
  mb={40}
  sx={{
    marginTop: "125px",
    p: {
      color: "orange.500",
      fontWeight: "bold"
    }
  }}
>
  <Text color="blue.500">
    I should be blue, but the sx props of my parent makes me orange!
  </Text>
</Box>
```

You can see a running CodeSandbox here: https://codesandbox.io/s/ecstatic-mountain-xo7iee?file=/src/index.tsx

**Further reading:** 

- [sx vs __css prop](https://github.com/chakra-ui/chakra-ui/discussions/5182) explains a bit more about the `sx` prop and `__css`. 
- [__css prop specificity](https://chakra-ui.com/docs/styled-system/theming/component-style#:~:text=It%20has%20the%20same%20API%20as%20the%20sx%20prop%2C%20but%20has%20a%20lower%20style%20priority.)

**Reply by nborzunov** [reply]
Thank you, @BillyPurvis for the solutions and the detailed explanation. It helped me a lot!

**Reply by BillyPurvis** [reply]
No problem! I'm glad it helped you! :)
